rep-0146.rst

```diff
@@ -0,0 +1,80 @@
+REP: 146
+Title: ROS Robot Parameter Naming Convention
+Author: Kei Okada <k-okada@jsk.t.u-tokyo.ac.jp>
+Status: Draft
+Type: Informational
+Content-Type: text/x-rst
+Created: 31-Jly-2015
+Post-History:
+
+Abstract
+========
+
+This REP aims to define the best practices for robot related parameter
+name space for making robot-agnostic packages.
+
+
+Motivation
+==========
+
+As the number of ROS-supported robot increases, we will write
+robot-agnostic packages, but still they need robot-specific
+information as calibration parameters.
+
+A `robot_description` parameter is widely shared naming convention
+among ROS community and this REP proposes such conventions for robot specific information on
+parameter, so application program can build their program on top of that.
+
+
+Specifications
+==============
+
+Parameters
+----------
+
+robot_description
+'''''''''''''''''
+
+`robot_description` stores robot kinematics / geometry information using
+URDF format.
+
+
+robot/type
+''''''''''
+
+`robot/type` stores robot type, for example 'pr2' in PR2 robot case [1]
+
+
+robot/name
+''''''''''
+
+`robot/name` stores robot name, default pr2 name is 'robot' [1]
+
+robot/info
```

Comment:
I don't think that this is currently defined well enough to be useful. I see the name and type as equivalents to `lsb_release` type of information. It would be easy to have a command line tool that just lists robots on the system and displays their name and type. But the info could be printed, but not much else.

If we want to keep this field we'd have to define some level of schema to make it useful.

Reply:
I added 'robot_info' because I'm afraid users willing to add more robot specific information depending on their use-cases, For example pepper robot can have characteristics of languages they speak or users who own the robot. But I do not want to consider/discuss if we should add `robot_language`, `robot_owner` , because that is very difficult and ambiguous and it is not clear if we should add these information as a  ROS-wide convention.

So 'robot_info'  will store information which each-users willing to store as their robot specific parameters.

Reply:
That's exactly my point. If it's going to be whatever the user wants to store, they have the entire parameter server to store anything they want however they want.  Setting these sort of standards is mostly about restricting what is consider proper usage of a key in the parameter server.

If we were to encourage unstructured use of `robot_info` now. In the future if we wanted to develop a standard schema for `robot_info` we would break every user who had already been using it in their own way, thus we'd actually have to then pick a different key for the new standard approach so as not to collide with the "old" standard which wasn't a standard.